tests/integration/cloud/clouds/test_ec2.py

```diff
@@ -9,115 +9,65 @@
 import yaml
 
 # Import Salt Libs
-from salt.config import cloud_providers_config
 import salt.utils.cloud
 import salt.utils.files
 import salt.utils.yaml
 
 # Import Salt Testing Libs
-from tests.support.case import ShellCase
 from tests.support.paths import FILES
-from tests.support.helpers import expensiveTest, generate_random_name
+from tests.support.helpers import expensiveTest
 from tests.support.unit import skipIf
 from tests.support import win_installer
 
-
 # Create the cloud instance name to be used throughout the tests
-INSTANCE_NAME = generate_random_name('CLOUD-TEST-')
-PROVIDER_NAME = 'ec2'
+from tests.integration.cloud.helpers.cloud_test_base import CloudTest
+
 HAS_WINRM = salt.utils.cloud.HAS_WINRM and salt.utils.cloud.HAS_SMB
+# THis test needs a longer timeout than other cloud tests
 TIMEOUT = 1200
 
 
-class EC2Test(ShellCase):
+def _fetch_installer():
     '''
-    Integration tests for the EC2 cloud provider in Salt-Cloud
+    Make sure the testing environment has a Windows installer executable.
     '''
+    # Determine the downloaded installer name by searching the files
+    # directory for the first file that looks like an installer.
+    for path, dirs, files in os.walk(FILES):
+        for file in files:
+            if file.startswith(win_installer.PREFIX):
+                return file
+    # Download the latest Windows installer executable
+    name = win_installer.latest_installer_name()
+    path = os.path.join(FILES, name)
+    with salt.utils.files.fopen(path, 'wb') as fp:
+        win_installer.download_and_verify(fp, name)
+    return name
 
-    def _installer_name(self):
-        '''
-        Determine the downloaded installer name by searching the files
-        directory for the firt file that loosk like an installer.
-        '''
-        for path, dirs, files in os.walk(FILES):
-            for file in files:
-                if file.startswith(win_installer.PREFIX):
-                    return file
-            break
-        return
-
-    def _fetch_latest_installer(self):
-        '''
-        Download the latest Windows installer executable
-        '''
-        name = win_installer.latest_installer_name()
-        path = os.path.join(FILES, name)
-        with salt.utils.files.fopen(path, 'wb') as fp:
-            win_installer.download_and_verify(fp, name)
-        return name
 
-    def _ensure_installer(self):
-        '''
-        Make sure the testing environment has a Windows installer executbale.
-        '''
-        name = self._installer_name()
-        if name:
-            return name
-        return self._fetch_latest_installer()
+INSTALLER = _fetch_installer()
```

Comment:
Does this need to be at the module global level?
Than means, for example, when pytest collects tests(I know 2019.2.1 does not run pytest, but the "problem" remains), this code will run, even if the test is to be skipped.
Stick it in `setUp` or `setUpClass`?

Reply:
That makes sense :+1: